infretis/core/tis.py

```diff
@@ -900,18 +905,25 @@ def retis_swap_zero(
     logger.info("Point is %s", phase_point.order)
     engine1.dump_phasepoint(phase_point, "second")
     path0.append(phase_point)
-    if path0.length == maxlen0:
-        path0.status = "BTX"
-    elif path0.length < 3:
-        path0.status = "BTS"
-    elif (
-        "L" not in set(ens_set0["start_cond"])
-        and "L" in path0.check_interfaces(ens_set0["interfaces"])[:2]
+
+    if (
+        (
+            "L" not in set(ens_set0["start_cond"])  # if not lambda_minus_one, 0- start_cond is ['R']
+            and "L" in path0.check_interfaces(ens_set0["interfaces"])[:2]
+        ) 
+        or (
+            "L" in set(ens_set0["start_cond"])      # if lambda_minus_one, 0- start_cond is ['L', 'R']
+            and "R" in set(ens_set0["start_cond"]) 
+            and "L" in path_old0.check_interfaces(ens_set0["interfaces"])[1:2] # reject 0- RML and LML swap with 0+
+        )
```

Comment:
I think we can simply do an early rejection before doing any MD steps at all in the case that path_old0 is either RML or LML

```python3
    # if lambda_minus_one, reject early if path_old0
    if set(ens_set0["start_cond"]) == set(["L", "R"]):
        if path_old0.check_interfaces(ens_set0["interfaces"])[1] == "L":
            return False, [path_old0, path_old1], "0-L"
```
e.g. like this and put it on before the line `    # 1. Generate path for [0^-] from [0^+]:`